Source/BestProjectEver/Private/Components/BPE_HealthComponent.cpp

```diff
@@ -0,0 +1,81 @@
+// Fill out your copyright notice in the Description page of Project Settings.
+
+
+#include "Components/BPE_HealthComponent.h"
+
+#include "Character/BPE_Enemy.h"
+#include "Net/UnrealNetwork.h"
+
+
+UBPE_HealthComponent::UBPE_HealthComponent()
+{
+	PrimaryComponentTick.bCanEverTick = false;
+
+	MaxHealth = 100.0f;
+	bIsDead = false;
+}
+
+//----------------------------------------------------------------------------------------------------------------------
+void UBPE_HealthComponent::BeginPlay()
+{
+	Super::BeginPlay();
+
+	CurrentHealth = MaxHealth;
+	AActor* ActorOwner = GetOwner();
```

Comment:
There's no harm in doing GetOwner directly in your calls. You're not saving any performance by caching it beforehand, since the GetOwner() call is inline and directly returns the owner.